src/gui/src/search.cpp

```diff
@@ -257,10 +257,22 @@ void Search::updateShapes(odb::dbBlock* block)
   data.snet_via_shapes_.clear();
   data.snet_shapes_.clear();
 
+  LayerMap<std::vector<RouteBoxValue<odb::dbNet*>>> net_shapes;
+  LayerMap<std::vector<SNetValue<odb::dbNet*>>> snet_shapes;
+  LayerMap<std::vector<SNetSBoxValue<odb::dbNet*>>> snet_net_via_shapes;
   for (odb::dbNet* net : block->getNets()) {
-    addNet(net);
-    addSNet(net);
+    addNet(net, net_shapes);
+    addSNet(net, snet_shapes, snet_net_via_shapes);
   }
+  for (const auto& [layer, layer_shapes] : snet_shapes) {
+    data.snet_shapes_[layer].insert(layer_shapes.begin(), layer_shapes.end());
```

Comment:
Rather than insert I think you need to call the ctor.  In drt it does:

```
  for (auto i = 0; i < numLayers; i++) {
    shapes_.at(i) = boost::move(RTree<frBlockObject*>(allShapes.at(i)));
```

Reply:
thanks.